lib/vsc/utils/generaloption.py

```diff
@@ -29,25 +29,31 @@
 @author: Stijn De Weirdt (Ghent University)
 @author: Jens Timmerman (Ghent University)
 """
-
-import ConfigParser
 import copy
 import difflib
 import inspect
 import operator
 import os
 import re
-import StringIO
 import sys
 import textwrap
+from functools import reduce
 from optparse import OptionParser, OptionGroup, Option, Values
 from optparse import BadOptionError, SUPPRESS_USAGE, OptionValueError
 from optparse import SUPPRESS_HELP as nohelp  # supported in optparse of python v2.4
 from optparse import gettext as _gettext  # this is gettext.gettext normally
+
+try:
+    from cStringIO import StringIO  # Python 2
+    import ConfigParser as configparser
+except ImportError:
+    from io import StringIO  # Python 3
+    import configparser
+
 from vsc.utils.dateandtime import date_parser, datetime_parser
 from vsc.utils.docs import mk_rst_table
 from vsc.utils.fancylogger import getLogger, setroot, setLogLevel, getDetailsLogLevels
-from vsc.utils.missing import shell_quote, nub
+from vsc.utils.missing import is_string, nub, shell_quote
```

Comment:
can we use the `is_string` for the fancylogger basestring replacements?

Reply:
done